webpack/components/extensions/HostDetails/Tabs/RepositorySetsTab/RepositorySetsTab.js

```diff
@@ -280,9 +284,17 @@ const RepositorySetsTab = () => {
 
   const hostDetailsStatus = useSelector(state => selectHostDetailsStatus(state));
 
+  // Ignore the toggle group when deciding if there is emptyContent. This will
+  // ensure the correct EmptyStateMessage
+  const isFiltering = activeFilters?.length &&
+    !isEqual(new Set(activeFilters), new Set(defaultFilters));
+  const emptyContent = (results && results.length === 0) && !searchQuery && !isFiltering;
+  const showPrimaryAction = (toggleGroupState === SHOW_ALL) && emptyContent;
+  const showSecondaryAction = showPrimaryAction;
+
   const resetFilters = () => {
     setStatusSelected(STATUS_LABEL);
-    setToggleGroupState(defaultToggleGroupState);
+    if (emptyContent) setToggleGroupState(SHOW_ALL);
```

Comment:
Would this method `resetFilters()` ever be called when there is some content with Limit to Environment?

Reply:
No, I think it's only called when the table is empty.